### DISCUSSION TITLE
ZLUDA Automatic1111 fast generation and slow output

### DISCUSSION THREAD

**Discussion by yadaweg** [lshqqytiger/stable-diffusion-webui-directml#538] [Q&A]
Why is my ZLUDA like that? Extremely fast 1-2 seconds generation, then outputting the picture for ~15 seconds
![image](https://github.com/user-attachments/assets/16cb3606-f1ce-4512-a048-6eb47c0a33db)

**Comment by ride5k** [comment]
i can confirm the discrepancy exists.  it does not happen if you use UNIPC sampler which provides "reasonable" iteration stats.  generation takes about the same total time vs other samplers leading me to believe it's a math/counter type of issue.

it does fuck up the preview window display, but other than that it's just an annoyance.

**Reply by yadaweg** [reply]
I'm gettin same 15 seconds on UniPC via slower iterations

**Comment by lshqqytiger** [comment]
It is synchronization issue. I had this same issue at the first time, but it was gone after several tries. SD.Next provides "dynamic" attentions that make synchronizations more strictly. It might help in some cases.

**Reply by yadaweg** [reply]
Will try that and give some feedback
UPD: so, basically, ZLUDA is not much faster than DirectML for my setup, BUT I couldn't run XL models w/ DirectML, like, at all, now it's running with no parameters smoothly
Imma try out it on my Linux Automatic1111 and SD.Next next
UPD2: I'm too stupid so Linux won't work for me
